### PR TITLE
Small code cleanups

### PR DESCRIPTION
**Pull request by jvz**
I figured I'd try to clean up some things while I was browsing some source files.

### Proposed changelog entries

* Ensure that `Jenkins#setViews()` updates the configuration only when all views are successfully set

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/java/jenkins/model/Jenkins.java [hunk 1]
```diff
@@ -1896,13 +1896,11 @@ public void addView(View v) throws IOException {
...
     // offers no such operation
     public void setViews(Collection<View> views) throws IOException {
-        BulkChange bc = new BulkChange(this);
-        try {
+        try (BulkChange bc = new BulkChange(this)) {
             this.views.clear();
             for (View v : views) {
                 addView(v);
             }
-        } finally {
```
Comment (daniel-beck) [inline core/src/main/java/jenkins/model/Jenkins.java:1905]: This doesn't look like it's the same, but it also looks like it might fix a bug…?

Comment (jvz) [reply core/src/main/java/jenkins/model/Jenkins.java:1905]: Oh snap, it does fix a bug.

Comment (oleg-nenashev) [reply core/src/main/java/jenkins/model/Jenkins.java:1905]: Uhm... Indeed there is a bug in the original logic which may lead to partial view data on the disk. Not fun. Probably even worth a changelog entry.

**Review by oleg-nenashev — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

**Review by StefanSpieker — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
Added the changelog entry

We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

**Review by timja — APPROVED** [review]

**Comment by jvz** [comment]
Thanks for updating the metadata!

File: core/src/main/java/jenkins/model/Jenkins.java (hunk 1 above)
Comment (daniel-beck) [reply core/src/main/java/jenkins/model/Jenkins.java:1905]: Well, why would there be a bulk change if you want to save partial data and never roll back?

**Pull request merged**